### ISSUE TITLE
How to run `npm install` in sub directory?

### ISSUE THREAD

**Issue by gocreating** [heroku/heroku#1869]
Heroku automatically runs `npm install` in the root directory

but I also want to install dependencies in sub directory

anybody knows how to do this?

**Comment by ransombriggs** [comment]
The toolbelt installs its own installation of node in the .heroku directory that it uses to manage the plugin architecture.  It would be best to leave that install of node alone and do a separate install of node if you wish to use it for node for one of your own projects.  Am I understanding your question properly?

**Comment by gocreating** [comment]
@ransombriggs Yeah that's my question. I think I should not install any dependency outside the root directory. It's not practical due to heroku's 60 seconds boot timeout. I am going to figure out another way, thank you!

**Comment by legarland** [comment]
I created a package.json in the root directory and used a postinstall script to change directories and run the npm install:
`"start": 
    { 
        "postinstall": "cd server && npm install" 
    }`
